grammars/jsdoc.cson

```diff
@@ -0,0 +1,478 @@
+      '3':
+        'name': 'variable.other.jsdoc'
+    'match': '((@)(?:default(?:value)?|license|tutorial|variation|version))\\s+([^\\s*]+)'
+  }
+  {
+    # Tags without arguments, or a tag without expected arguments. Because JSDoc permits
+    # tags to be spread across lines, we should at least highlight the opening tag for
+    # stuff like this:
+    #
+    #    /**
+    #      * @param
+    #      * {type}
+    #      * name
+    'captures':
+      '1':
+        'name': 'punctuation.definition.block.tag.jsdoc'
+    'match': '''(?x) (@)
+      (?:abstract|access|alias|arg|argument|async|attribute|augments|author|beta|borrows|bubbles
+      |callback|chainable|class|classdesc|code|config|const|constant|constructor|constructs|copyright
+      |default|defaultvalue|define|deprecated|desc|description|dict|emits|enum|event|example|exception
+      |exports?|extends|extension(?:_?for)?|external|externs|file|fileoverview|final|fires|for|func
+      |function|global|host|ignore|implements|implicitCast|inherit[Dd]oc|inner|instance|interface|kind
+      |lends|license|listens|main|member|memberof!?|method|mixes|mixins?|modifies|module|name|namespace
+      |noalias|nocollapse|nocompile|nosideeffects|override|overview|package|param|preserve|private|prop
+      |property|protected|public|read[Oo]nly|record|require[ds]|returns?|see|since|static|struct|submodule
+      |summary|suppress|template|this|throws|todo|tutorial|type|typedef|unrestricted|uses|var|variation
+      |version|virtual|writeOnce)
+      \\b
+    '''
+    'name': 'storage.type.class.jsdoc'
+  }
+  {
+    # Description preceding {@inline tag}
+    'captures':
+      '1':
+        'name': 'punctuation.definition.bracket.square.begin.jsdoc'
+      '2':
+        'name': 'punctuation.definition.bracket.square.end.jsdoc'
+    'match': '(\\[)[^\\]]+(\\])(?={@(?:link|linkcode|linkplain|tutorial))'
+    'name': 'constant.other.description.jsdoc'
+  }
+  {
+    # {@link|tutorial …}
+    'begin': '({@)(?:link|linkcode|linkplain|tutorial)\\s*'
+    'beginCaptures':
+      '0':
+        'name': 'storage.type.class.jsdoc'
+      '1':
+        'name': 'punctuation.definition.inline.tag.begin.jsdoc'
+    'end': '}|(?=\\*/)'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.definition.inline.tag.end.jsdoc'
+    'name': 'entity.name.type.instance.jsdoc'
+    'patterns': [
+      {
+        'captures':
+          '1':
+            'name': 'variable.other.link.underline.jsdoc'
+          '2':
+            'name': 'punctuation.separator.pipe.jsdoc'
+        'match': '\\G((?=https?://)(?:[^|}\\s*]|\\*[/])+)(\\|)?'
+      }
+      {
+        'captures':
+          '1':
+            'name': 'variable.other.description.jsdoc'
+          '2':
+            'name': 'punctuation.separator.pipe.jsdoc'
+        'match': '\\G((?:[^{}@\\s|*]|\\*[^/])+)(\\|)?'
+      }
+    ]
+  }
+]
+
+'repository':
+
+  # Balanced brackets (square or curly)
+  'brackets':
```

Comment:
This is only used to keep stuff balanced inside default values. I chose not to add any scopes to their contents so it was clearer to the reader where the JSDoc token ended.